source/index-management/index-management.rst

```diff
@@ -852,6 +852,181 @@ Example::
+    */
+    createSearchIndex(model: SearchIndexModel, options: Optional<CreateSearchIndexOptions>): String;
+
+    /**
+    * Creates multiple search indexes on the collection.
+    * 
+    * @return An iterable of the newly created index names.
+    */
+    createSearchIndexes(models: Iterable<SearchIndexModel>, options: CreateSearchIndexOptions): Iterable<String> ;
+
+    /**
+    * Updates the search index with the given name to use the provided 
+    * definition.
+    */
+    updateSearchIndex(name: String, definition: Document, options: Optional<UpdateSearchIndexOptions>): void;
+
+    /**
+    * Drops the search index with the given name.
+    */
+    dropSearchIndex(name: String, options: Optional<DropSearchIndexOptions>): void;
+
+    /**
+    * Gets index information for all search indexes in the collection.
+    */
+    listSearchIndexes(name: Optional<String>, aggregationOptions: Optional<AggregationOptions>, listIndexOptions: Optional<ListSearchIndexOptions>): Cursor<{ name: String }>;
+  }
+
+Index View API for Search Indexes
+---------------------------------
+
+.. code:: typescript
+
+  interface Collection {
+    /**
+     * Returns the search index view for this collection.
+     */
+    searchIndexes(options: Optional<ListSearchIndexesOptions>): SearchIndexView;
+  }
+
+  interface SearchIndexView extends Iterable<Document> {
+    /**
+     * Enumerates the index information for all indexes in the collection. This should be
+     * implemented as described in the :ref:`Enumerate Indexes` section, although the naming
+     * requirement is dropped in favor of the driver language standard for handling iteration
+     * over a sequence of objects.
+     *
+     * @see https://github.com/mongodb/specifications/blob/master/source/enumerate-indexes.rst
+     *
+     * @note For drivers that cannot make the IndexView iterable, they MUST implement a list
+     *   method. See below.
+     */
+    iterator(): Iterator<{ name: string }>;
+
+    /**
+     * For drivers that cannot make SearchIndexView iterable, they MUST implement this method to
+     * return a list of indexes. In the case of async drivers, this MAY return a Future<Cursor>
+     *  or language/implementation equivalent.
+     */
+    list(): Cursor<{ name: String }>;
+
+    /**
+     * This is a convenience method for creating a single index.
+     *
+     * @return The name of the created index.
+     *
+     * @note Drivers MAY opt to implement this method signature, the signature that
+     *   takes an IndexModel as a parameter, or for those languages with method
+     *   overloading MAY decide to implement both.
+     *
+     * @note Drivers MAY combine the two options types into a single one. If the options are
+     *   explicitly typed, the combined options type MUST be named CreateOneIndexOptions or an acceptable
+     *   variation.
+     */
+    createOne(definition: SearchIndexDefinition, options: CreateSearchIndexOptions): String;
```

Comment:
I think you want to specify `name: string, definition: Document` and provide an optional overload with a `SearchIndexModel` just as the standard API allows. `SearchIndexDefinition` is not defined in the spec changes, and I think that's intentional for future-proofing. If not, we should define what those fields are and replace the usages of `Document` in the changes. The syntax doc looks like:

```js
{
  analyzer: "<analyzer for index>",
  searchAnalyzer: "<analyzer for query>",
  analyzers: [<custom analyzer>],
  storedSource: [<boolean | stored source definition>],
  synonyms: [<synonym mapping>],
  mappings: {
    dynamic: true
  }
}
```

Reply:
I updated the references to `SearchIndexDefinition` to be `Document` instead, that was a mistake.  

I did intentional leave the definition as a `Document` for future compatibility.